File: mobile-app/lib/services/auth_service.dart
// ============================================================
// File: mobile-app/lib/services/auth_service.dart
// ============================================================

import 'api_service.dart';
import 'token_storage.dart';

class AuthService {
  static Future<Map<String, dynamic>> login(String email, String password) async {
    final res  = await ApiService.post('/auth/login', {'email': email, 'password': password});
    final data = res['data'] as Map<String, dynamic>;
    await TokenStorage.saveToken(data['accessToken'] as String);
    await TokenStorage.saveUser(data['user'] as Map<String, dynamic>);
    return data;
  }

  static Future<Map<String, dynamic>> register(String name, String email, String password) async {
    final res  = await ApiService.post('/auth/register', {'name': name, 'email': email, 'password': password});
    final data = res['data'] as Map<String, dynamic>;
    await TokenStorage.saveToken(data['accessToken'] as String);
    await TokenStorage.saveUser(data['user'] as Map<String, dynamic>);
    return data;
  }

  static Future<void> logout() async {
    try { await ApiService.post('/auth/logout', {}); } catch (_) {}
    await TokenStorage.clear();
  }

  static Future<String?> getToken()               => TokenStorage.getToken();
  static Future<Map<String, dynamic>?> getUser()  => TokenStorage.getUser();
}

File: frontend-web/src/services/authService.js
// File: frontend-web/src/services/authService.js
import api from './api'
export const authService = {
  login:    (email, password) => api.post('/auth/login',    { email, password }).then(r => r.data.data),
  register: (name, email, password) => api.post('/auth/register', { name, email, password }).then(r => r.data.data),
  logout:   () => api.post('/auth/logout'),
  getMe:    () => api.get('/auth/me').then(r => r.data.data),
}
